tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -142,6 +143,43 @@
 import argparse
 import configargparse
 
+
+# attempt to import chplcheck
+try:
+    chpl_home = os.environ.get("CHPL_HOME")
+    if chpl_home is None:
+        raise ValueError("CHPL_HOME not set")
+    chplcheck_path = os.path.join(chpl_home, "tools", "chplcheck", "src")
+    # Add chplcheck to the path, but load via importlib
+    sys.path.append(chplcheck_path)
+
+    def load_module(module_name: str):
+        file_path = os.path.join(chplcheck_path, module_name + ".py")
+        spec = importlib.util.spec_from_file_location(module_name, file_path)
+        if spec is None:
+            raise ValueError(f"Could not load module from {file_path}")
+        module = importlib.util.module_from_spec(spec)
+        sys.modules[module_name] = module
+        if spec.loader is None:
+            raise ValueError(f"Could not load module from {file_path}")
+        spec.loader.exec_module(module)
+        return module
+
+    chplcheck = load_module("chplcheck")
+    chplcheck_lsp = load_module("lsp")
+    chplcheck_config = load_module("config")
+    chplcheck_driver = load_module("driver")
+    chplcheck_rules = load_module("rules")
+
+except ValueError as e:
+    if os.environ.get("CHPL_DEVELOPER", None):
+        print("Error loading chplcheck: ", str(e), file=sys.stderr)
+    chplcheck = None
+    chplcheck_lsp = None
+    chplcheck_config = None
+    chplcheck_driver = None
+    chplcheck_rules = None
+
```

Comment:
I suggest you wrap this in a class that contain the various modules. Something like this:

```Python
class ChplcheckProxy:
    def __init__(self):
        chpl_home = ...
        ...
        self.main = self.load_module("chplcheck")
        self.lsp = self.load_module("lsp")

chplcheck = None
try:
    chplcheck = ChplcheckProxy()
```
A few of things:

1.  Note that I've renamed the variables to avoid having to prefix them with the name chplcheck internally. Externally, you'd access them like `chplcheck.main` or `chplcheck.lsp`.
2. if the only errors that can occur are the `ValueError`s that you raise, consider just making this a static method and returning `None` to avoid the `none-try-some` in the main body.
3. Ideally, you'd only execute this if `--chplcheck` was thrown, but then you'd need to have some kind of two-phase parsing.

Reply:
I've added a proxy object

> Ideally, you'd only execute this if --chplcheck was thrown, but then you'd need to have some kind of two-phase parsing.

I think this is the ideal from a purity standpoint, but maybe not the most practical. Its possible to do `parse_known_args`, and then do some extra parsing of the command line arguments. But this adds extra complexity to our parsing that I don't want to have to maintain, and I am not sure there is much benefit